internal/controller/instaslice_controller.go

```diff
@@ -68,17 +70,42 @@ const requeueDelay = 2 * time.Second
 //+kubebuilder:rbac:groups=inference.codeflare.dev,resources=instaslices/finalizers,verbs=update
 //+kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups="",resources=nodes,verbs=get;list;update;patch
+//+kubebuilder:rbac:groups=apps,resources=daemonsets,verbs=get;list;watch;create;update;delete
 
 func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 
+	// 1. Ensure DaemonSet is deployed
+	daemonSet := &appsv1.DaemonSet{}
+	err := r.Get(ctx, types.NamespacedName{Name: "instaslice-operator-controller-daemonset", Namespace: "instaslice-operator-system"}, daemonSet)
```

Comment:
The name and namespace should probably be constants. They are used here and in the createInstasliceDaemonset